### PR TITLE
Add possibility to specify user via mail subject "u-" + Add basic Docker implementation

### LINKED ISSUE #18: Add user to card
Is it possible to add a user to a card that is mailed in?
In Trello this is done by writing ` @username` into the email subject line.
I just read the readme and it doesn't say there if this works or not.
We are currently considering switching from Trello to Deck and this is one of the crucial features.

Thanks for your help!

### PR DESCRIPTION
**Pull request by matbgn**
Fixes #18 

Take precedence on ASSIGN_SENDER to be able to create task for other users than mail sender.

### CODE REVIEW

File: lib/DeckClass.php [hunk 1]
```diff
@@ -88,9 +93,11 @@ public function addCard($data, $user, $board = null) {
             $card = $this->apiCall("POST", NC_SERVER . "/index.php/apps/deck/api/v1.0/boards/{$params->board}/stacks/{$params->stack}/cards", $data);
             $card->board = $params->board;
             $card->stack = $params->stack;
-    
+
+            if ($params->userId !== null) $user->userId = $params->userId;
```
Comment (oneWaveAdrian) [inline lib/DeckClass.php:97]: It looks like you are only trying to make sure `userId` is not exactly `null` on `$params`. I assume you just want to make sure it exists, therefore you could do it without the type check?
```suggestion
            if ($params->userId) $user->userId = $params->userId;
```

File: lib/DeckClass.php [hunk 2]
```diff
@@ -88,9 +93,11 @@ public function addCard($data, $user, $board = null) {
...
+
             if($this->responseCode == 200) {
-                if(ASSIGN_SENDER) $this->assignUser($card, $user);
+                if(ASSIGN_SENDER || $params->userId !== null) $this->assignUser($card, $user);
```
Comment (oneWaveAdrian) [inline lib/DeckClass.php:100]: It looks like you are only trying to make sure userId is not exactly null on $params. I assume you just want to make sure it exists, therefore you could do it without the type check?

```suggestion
                if(ASSIGN_SENDER || $params->userId) $this->assignUser($card, $user);
```

**Review by oneWaveAdrian — COMMENTED** [review]
Looks good to me so far, thanks for taking the time to do it!
Just a couple things I noticed and marked. Feel free to accept/decline, in the end it is your code :)

File: lib/DeckClass.php (hunk 1 above)
Comment (matbgn) [reply lib/DeckClass.php:97]: Thanks for your precious feedback I'm totally new to PHP so I was not confident to do this like that and Googled for the best approach. Again many thanks

**Comment by matbgn** [comment]
Again thank you for your feedback especially on the README where I made some fatigue errors.

**Comment by lucianpricop** [comment]
Thank you very much for your contribution! :)

**Pull request merged**